packages/go_router/lib/src/misc/refresh_stream.dart

```diff
@@ -22,12 +18,12 @@ import 'go_router.dart';
 /// );
 /// ```
 /// {@end-tool}
-class GoRouterRefreshStream extends ChangeNotifier {
-  /// Creates a [GoRouterRefreshStream].
+class StreamListenable extends ChangeNotifier {
+  /// Creates a [StreamListenable].
   ///
   /// Every time the [stream] receives an event the [GoRouter] will refresh its
   /// current route.
-  GoRouterRefreshStream(Stream<dynamic> stream) {
+  StreamListenable(Stream<dynamic> stream) {
```

Comment:
Isn't this a breaking change? What made us decide to rename this? Should we update the changelog to call this out?

Reply:
Good catch, I didn't intend to make breaking changes here - I'll change this back. 

I would like to deprecate it or at the very least add some tests. For example, what happens when the stream is cancelled? Does this class work with single-subscription streams, or just broadcast streams? This class isn't really related to routing either.